### ISSUE TITLE
Additional file created with -E suffix after the replace is done

### ISSUE THREAD

**Issue by cksidharthan** [nvim-pack/nvim-spectre#249]
I have the latest version of nvim-spectre. and I tried to search and replace a string in my current file, I see that the text is replaced, but there is an additional file that is created after the replace is done with a -E suffix. 

For example, when i try to replace a string in a file named `test.go`, i see that the text replace is done correctly in the file, but there is one more file that is being created as `test.go-E`

I have the default config without an custom config or something like below

```
return {
	"nvim-pack/nvim-spectre",
	dependencies = {
		"nvim-lua/plenary.nvim",
	},
        lazy = true,
	keys = {
		{
			"<leader>Rr",
			function()
				require("spectre").toggle()
			end,
			desc = "Toggle search and replace",
		},
		{
			"<leader>Rw",
			mode = { "v", "n" },
			function()
				require("spectre").open_visual({ select_word = true })
			end,
			desc = "Search word",
		},
		{
			"<leader>Rf",
			function()
				require("spectre").open_file_search({ select_word = true })
			end,
			desc = "Search in current file",
		},
	},
}

```

OS: MacOS
Platform ARM64
Neovim Version: 0.10

**Comment by NullVoxPopuli** [comment]
I'm running in to this as well

**Comment by srodrigo** [comment]
Having the same issue as well.

**Comment by alex-heinz** [comment]
same here: MacOS `14.6.1` on M3, Neovim `0.10.1`, gnu-sed `4.9`, nvim-spectre revision `ba7fb77`

**Comment by dimakiyaev** [comment]
helped me -  https://github.com/nvim-pack/nvim-spectre/issues/118

**Comment by cksidharthan** [comment]
even with this #118 fix, I'm getting those -E files 😢

**Comment by cksidharthan** [comment]
fixed by using solution in #118
